tests/HttpAdapterFactoryTest.php

```diff
@@ -56,6 +56,14 @@ public function testRegisterWithInvalidClass()
         HttpAdapterFactory::register('foo', 'stdClass');
     }
 
+    public function testGuess()
```

Comment:
Can you introduce a data provider and test different use cases instead of just a single one?